### PR TITLE
Encode database names and document Ids used n Urls

### PR DESCRIPTION
**Pull request by urdeveloper**
Spiegel doesn't work as expected when there's forward slash in the database name and document Ids.

### CODE REVIEW

**Comment by redgeoff** [comment]
@urdeveloper, thanks for the PR! This looks good on the surface. Can you:

1. Add at least one new test case, e.g. for DB.prototype._create that uses creates a DB with slash in the name?
2. Run `npm run beautify` and then commit (see https://github.com/redgeoff/slouch/blob/master/TESTING.md#beautify for more info)

File: scripts/db.js [hunk 1]
```diff
@@ -110,22 +110,22 @@ DB.prototype.changes = function (dbName, params) {
...
-    url: this._slouch._url + '/' + dbName + '/' + viewDocId + '/_view/' + view,
+    url: this._slouch._url + '/' + encodeURIComponent(dbName) + '/' + viewDocId + '/_view/' + view,
     qs: params
   }, 'rows.*');
 };
 
 DB.prototype.viewArray = function (dbName, viewDocId, view, params) {
   return this._slouch._req({
-    url: this._slouch._url + '/' + dbName + '/' + viewDocId + '/_view/' + view,
+    url: this._slouch._url + '/' + encodeURIComponent(dbName) + '/' + viewDocId + '/_view/' + view,
```
Comment (redgeoff) [inline scripts/db.js]: Does `viewDocId` need to be encoded as well?

File: scripts/db.js [hunk 2]
```diff
@@ -110,22 +110,22 @@ DB.prototype.changes = function (dbName, params) {
...
+    url: this._slouch._url + '/' + encodeURIComponent(dbName) + '/_changes',
     qs: params,
     parseBody: true
   });
 };
 
 DB.prototype.view = function (dbName, viewDocId, view, params) {
   return new CouchPersistentStreamIterator({
-    url: this._slouch._url + '/' + dbName + '/' + viewDocId + '/_view/' + view,
+    url: this._slouch._url + '/' + encodeURIComponent(dbName) + '/' + viewDocId + '/_view/' + view,
```
Comment (redgeoff) [inline scripts/db.js]: Does viewDocId need to be encoded as well?

File: scripts/db.js (hunk 1 above)
Comment (urdeveloper) [reply scripts/db.js]: Probably yes. I'll make the changes.

**Comment by redgeoff** [comment]
Looks like there are some JS errors: https://circleci.com/gh/redgeoff/slouch/176?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link. Currently, slouch requires ES5

**Comment by urdeveloper** [comment]
Failed while creating couchdb!

**Comment by redgeoff** [comment]
Yeah, sorry this is an annoying CircleCI thing that happens periodically. (It will be fixed by https://github.com/redgeoff/slouch/issues/56). For now, I've restarted the tests.

**Comment by urdeveloper** [comment]
There seems to be another couchdb process hanging on there.

**Comment by redgeoff** [comment]
Yeah, CircleCI decides to sometimes run its own instance of CouchDB based on some inconsistent logic :(. Restarted it again... this usually works.

**Comment by redgeoff** [comment]
@urdeveloper this looks great! Thanks for all the hard work. I'm going to merge this now and then create new Slouch and Spiegel builds.

**Comment by urdeveloper** [comment]
@redgeoff You're very welcome. It was fun!

**Comment by redgeoff** [comment]
@urdeveloper changes in [v0.1.0](https://github.com/redgeoff/spiegel/releases/tag/v0.1.0) and there is a new docker image.

**Pull request merged**